### PR TITLE
Remove the UpdateSplitter Actor.

### PR DESCRIPTION
**Pull request by fasaxc**
This improves performance by significantly reducing the number of
messages we pass around.

* [x] Finish UTs

### CODE REVIEW

**Comment by plwhite** [comment]
LGTM, but not merging while the comment at the top says "needs more UTs"!

**Comment by fasaxc** [comment]
Yeh, I lost the coverage of cleanup in the change.

File: calico/felix/splitter.py [hunk 1]
```diff
@@ -128,12 +151,10 @@ def on_endpoint_update(self, endpoint_id, endpoint):
...
         _log.debug("Endpoint update for %s.", endpoint_id)
-        for ipset_mgr in self.ipsets_mgrs:
-            ipset_mgr.on_endpoint_update(endpoint_id, endpoint, async=True)
-        for endpoint_mgr in self.endpoint_mgrs:
-            endpoint_mgr.on_endpoint_update(endpoint_id, endpoint, async=True)
+        for mgr in self.ep_upd_mgrs:
+            mgr.on_endpoint_update(endpoint_id, endpoint, async=True)
 
-    @actor_message()
     def on_ipam_pool_update(self, pool_id, pool):
```
Comment (alexwlchan) [inline calico/felix/splitter.py]: While you’re here, want to add a docstring to this function?

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
Fixed the UT failure and pushed the new revision.

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
UT hit an intermittent failure but the FVs (including their UT run) passed so I think this can go in...

**Pull request merged**